docs/security.rst

```diff
@@ -60,3 +60,70 @@ Girder returns an error when a Cross-Origin request is made (one with the
 ``Origin`` header) that does not match the system configuration settings.
 Although most modern web browsers also enforce this, some additional security
 is added by enforcing it at the request level.
+
+
+Database Injection Attacks
+--------------------------
+
+Girder defends against database injection attacks by using PyMongo as the only
+pathway between the application server and the database server. PyMongo--like
+most MongoDB client libraries build up a BSON object before sending the query
+to the database. This prevents most injection attacks. See the
+`MongoDB Documentation
+<http://docs.mongodb.org/manual/faq/developers/#how-does-mongodb-address-sql-or-query-injection>`_
+for more information.
+
+Additionally, we strongly recommend running MongoDB with JavaScript disabled
+unless explicitly needed for your Girder-based application or plugin. Again,
+see the `MongoDB Documentation
+<http://docs.mongodb.org/manual/faq/developers/#javascript>`_
+for more information.
+
+
+Session Management
+------------------
+
+Girder uses session management performed through the Girder-Token header or
+through a token passed through a GET parameter. This token is passed through
+the cookie and expires after a configurable amount of time. In order to prevent
+session stealing, it is highly recommended to run Girder under HTTPS.
+
+
+Cross-Site Scripting (XSS)
+--------------------------
+
+In order to protect against XSS attacks, all input from users is sanitized
+before presentation of the content on each page. This is handled by the
+template system Girder uses (`Jade <http://jade-lang.com/>`_). This sanitizes
+user-provided content.
+
+
+Cross-Site Request Forgery (CSRF)
+---------------------------------
+
+To prevent CSRF attacks, Girder requires the Girder-Token parameter as a header
+for all state-changing requests. This token is taken from the user's cookie and
+then passed in the request as part of the Girder one-page application and other
+clients such that the cookie alone is not enough to form a valid request. A
+sensible CORS policy (discussed above) also helps mitigate this attack vector.
+
+
+Dependent Libraries
+------------------
+
+Another common attack vector is through libraries upon which girder depends
+such as CherryPy, Jade, PyMongo, etc. During development, and before releases
```

Comment:
I disagree. An upstream change doesn't have to malicious to break things, since dependant library code is always designed to be used in conjunction with your own. Blindly accepting updates for your OS may be ok, because it's designed to work as a self-contained system. Blindly accepting new versions of libraries is a great way to break your own software due to API changes and such. Security bugs can be part of such breakages.